org.eclipse.tm4e.ui/src/main/java/org/eclipse/tm4e/ui/themes/ColorManager.java

```diff
@@ -47,4 +49,72 @@ public void dispose() {
 			e.next().dispose();
 		}
 	}
+
+	/**
+	 * Get the color from preferences store using a token.
+	 * 
+	 * @param tokenId
+	 *            name of the token
+	 * @return Color matching token
+	 */
+	public Color getPreferenceEditorColor(String tokenId) {
+		return getColor(stringToRGB(PreferenceUtils.getEditorsPreferenceStore().get(tokenId, "")));
+	}
+
+	/**
+	 * The method verifies that a color is defined in a preferences store using a token.
+	 * 
+	 * @param tokenId
+	 *            name of the token
+	 * @return color is user defined or not
+	 */
+	public boolean isColorUserDefined(String tokenId) {
+		return !PreferenceUtils.getEditorsPreferenceStore().getBoolean(getSystemDefaultToken(tokenId), true);
```

Comment:
Could we implement this without introducing a "SytemDefault"-based approach? Something like assuming that if a token is not part of the preferences, it means it's system default?

Reply:
Also, shouldn't be stored in another prefernece store than the default Editor one? As only TM4E uses them, it would better have them in TM4E preference store IMO.

Reply:
Do you suggest to store default values for all 5 variables used here and check them when it needs?

Reply:
Ok, I see how SytemDefault is used in other places of Platform and your usage is consistent with it.
You can ignore my previous comments.